artiq/firmware/ksupport/eh_artiq.rs

```diff
@@ -352,3 +361,29 @@ pub fn get_exception_id(name: &str) -> u32 {
     unimplemented!("unallocated internal exception id")
 }
 
+/// Takes as input exception id from host
+/// Generates a new exception with:
+///   * `id` set to `exn_id`
+///   * `message` set to corresponding exception name from `EXCEPTION_ID_LOOKUP`
+///
+/// The message is matched on host to ensure correct exception is being referred 
+/// This test checks the synchronization of exception ids for runtime errors
+#[no_mangle]
+pub extern "C-unwind" fn test_exception_id_sync(exn_id: u32) {
+    let message = EXCEPTION_ID_LOOKUP
+        .iter()
+        .find_map(|&(name, id)| if id == exn_id { Some(name) } else { None })
+        .unwrap_or("unallocated internal exception id");
+    
+    let exn = Exception {
+        id:       exn_id,
+        file:     file!().as_c_slice(),
+        line:     line!(),
+        column:   column!(),
```

Comment:
> I think column is pretty meaningless here.

For the id sync tests, the fields `file`, `line`, `column`, `function` and `param` are all not used. They are defined here because they were part of the Exception struct. Otherwise, the column is used to indicate the error location similar to Python trace tree.

Reply:
> For the macro it might be different, did you test it?

For macro, it will be expanded first and then evaluated. For instance calling `raise!` (defined in `firmware/lib.rs`) from `ksupport/nrt_bus` will cause `file!` in the macro to return `nrt_bus`. So, `file!` and `line!` are giving somewhat useful information. The bigger concern is the function name which is fixed to `(RUST Function)` for all cases. However, as these are not part of exception syncing, will create a separate PR for these.